rust-runtime/aws-smithy-http-server/src/protocol/mod.rs

```diff
@@ -143,15 +137,15 @@ pub fn accept_header_classifier(headers: &HeaderMap, content_type: &mime::Mime)
                  */
                 .flat_map(|s| s.split(',').map(|typ| typ.split(';').next().unwrap().trim()))
         })
-        .filter_map(|h| h.parse::<mime::Mime>().ok())
+        .filter_map(|h| mediatype::MediaType::parse(h).ok())
         .any(|mim| {
-            let typ = content_type.type_();
-            let subtype = content_type.subtype();
+            let typ = content_type.ty;
+            let subtype = content_type.subty;
             // Accept: */*, type/*, type/subtype
-            match (mim.type_(), mim.subtype()) {
-                (t, s) if t == typ && s == subtype => true,
-                (t, mime::STAR) if t == typ => true,
-                (mime::STAR, mime::STAR) => true,
+            match (mim.ty.as_str(), mim.subty.as_str()) {
+                (t, s) if typ == t && subtype == s => true,
+                (t, "*") if typ == t => true,
+                ("*", "*") => true,
```

Comment:
`mediatype` has constants too

Reply:
Yea originally my match statement looked like 
```
            match (mim.ty, mim.subty) {
                (t, s) if typ == t && subtype == s => true,
                (t, mediatype::names::_STAR) if typ == t => true,
                (mediatype::names::_STAR, mediatype::names::_STAR) => true,
                _ => false,
            }
```
but this does not compile. You can't use mediatype constants in a match statement or else you get this error:
```
error: to use a constant of type `mediatype::Name<'_>` in a pattern, `mediatype::Name<'_>` must be annotated with `#[derive(PartialEq, Eq)]`
   --> aws-smithy-http-server/src/protocol/mod.rs:147:21
    |
147 |                 (t, mediatype::names::_STAR) if typ == t => true,
    |                     ^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: the traits must be derived, manual `impl`s are not sufficient
    = note: see https://doc.rust-lang.org/stable/std/marker/trait.StructuralEq.html for details
```
    
Our other option is to just use an if-else:
    
```
            if mim.ty == typ {
                mim.subty == subtype || mim.subty == mediatype::names::_STAR
            } else {
                mim.ty == mediatype::names::_STAR && mim.subty == mediatype::names::_STAR
            }
```

I prefer the way I wrote in the PR. I think it's more readable. But I'd be happy to change it to the if-else way, if that's what y'all want.